src/functions.jl

```diff
@@ -492,4 +492,51 @@ function LinearAlgebra.vecdot(x::AbstractArray{Variable}, y::AbstractArray{Linea
     vecdot!(zero(QuadraticFunction{T}), x, y)
 end
 
+# vcat
+function _vcat!(dest, i::Integer, arg::T, args::Vararg{<:Any, N}) where {N, T}
+    @boundscheck i >= firstindex(dest) && (i + length(arg) - 1) <= lastindex(dest) || throw(DimensionMismatch())
+    for x in arg
```

Comment:
Move `@inbounds` in front of `for`, otherwise there are still bounds checks for iterating over `arg`.

Reply:
Oh, didn't know that, cool